backend/src/main/java/com/bakdata/conquery/sql/conversion/forms/PostgresStratificationFunctions.java

```diff
@@ -63,14 +72,24 @@ public Field<Date> yearEndQuarterAligned(ColumnDateRange dateRange) {
 
 	@Override
 	public Field<Date> quarterStart(ColumnDateRange dateRange) {
-		Field<Integer> quarter = functionProvider.extract(DatePart.QUARTER, lower(dateRange));
-		return addQuarters(jumpToYearStart(lower(dateRange)), quarter, Offset.MINUS_ONE);
+		return jumpToQuarterStart(lower(dateRange));
+	}
+
+	@Override
+	protected Field<Date> jumpToQuarterStart(Field<Date> date) {
+		Field<Integer> quarter = functionProvider.extract(DatePart.QUARTER, date);
+		return addQuarters(jumpToYearStart(date), quarter, Offset.MINUS_ONE);
 	}
 
 	@Override
 	public Field<Date> quarterEnd(ColumnDateRange dateRange) {
-		Field<Timestamp> yearStart = dateTruncate(DSL.val("year"), upper(dateRange));
-		Field<Date> quarterEndInclusive = upper(dateRange).minus(1);
+		return jumpToNextQuarterStart(upper(dateRange));
```

Comment:
links hattest du auch noch minus 1, kannst du bitte einen kommentar dran machen, wrum du das nicht brauchst?

Reply:
Ich brauche sie, hatte es allerdings an der falschen Stelle in der jumpToNextQuarterStart() Methode.